plugins/http2_adapter/lib/src/connection_manager_imp.dart

```diff
@@ -68,16 +68,20 @@ class _ConnectionManager implements ConnectionManager {
     late SecureSocket socket;
     try {
       // Create socket
-      socket = await SecureSocket.connect(
-        uri.host,
-        uri.port,
-        timeout: options.connectTimeout > 0
-            ? Duration(milliseconds: options.connectTimeout)
-            : null,
-        context: clientConfig.context,
-        onBadCertificate: clientConfig.onBadCertificate,
-        supportedProtocols: ['h2'],
-      );
+      if (clientConfig.proxy != null) {
+        socket = await _createSocketWithProxy(
+          uri,
+          options,
+          clientConfig,
+          clientConfig.proxy!,
```

Comment:
Since you are passing the `clientConfig` anyways, both cases can actually be collapsed into one method call and do the `if (clientConfig.proxy != null)` check in there.

Reply:
fixed and aded proxy authorization